src/cache.py

```diff
@@ -0,0 +1,121 @@
+"""
+|==========================================|
+| Copyright (c) 2017-present Reece Dunham  |
+| All rights reserved                      |
+| Used with explicit permission            |
+|==========================================|
+"""
+
+# Taken from a project that may become open source soon, please keep the copyright header
+
+from io import TextIOWrapper
+
+
+class AbstractFile:
+    """Used for defining a file"""
+    def __init__(self, name: str):
+        """
+        Create class instance
+
+        :param name: the name of the file (and relative path if needed)
+        """
+        self.name = name
+
+    def get_name(self) -> str:
+        """
+        Get the file's name
+
+        :return: the file's name
+        :rtype: str
+        """
+        return self.name
+
+
+class FileCache:
+    """Allows for caching of files easily!"""
+    def __init__(self):
+        """
+        Creates the class
+
+        :return: nothing
+        """
+        self.file_caches = []
+
+    def refresh(self, file: AbstractFile) -> (object, list):
+        """
+        Refreshes the @TextIOWrapper for the file
+
+        :param file: the @AbstractFile to refresh
+        :return: some list thing
+        """
+        with open(file.get_name(), mode="r") as filehandler:
```

Comment:
```
Traceback (most recent call last):
  File "app.py", line 18, in <module>
    replied = fileLoadCache.refresh(replied_file.get_name())
  File "cache.py", line 51, in refresh
    with open(file.get_name(), mode="r") as filehandler:
AttributeError: 'str' object has no attribute 'get_name'
```

Reply:
oh right smh

Reply:
which python version?

Reply:
2.7, but I can upgrade if it's required

Reply:
sir
your going to need latest

Reply:
I can run 3.6.7

Reply:
3.7.2 is prob best

Reply:
Hmm.. would 3.7.1 work?

Reply:
@RDIL 3.7.2 is now installed on the server

Reply:
Yeah that should work - thanks!